RxSwift/Observables/Breakpoint.swift

```diff
@@ -0,0 +1,182 @@
+     - parameter onSubscribed: Predicate to invoke after subscribing to source observable sequence. Return `true` from this closure to raise `SIGTRAP`, or false to continue.
+     - parameter onDispose: Predicate to invoke after subscription to source observable has been disposed for any reason. It can be either because sequence terminates for some reason or observer subscription being disposed. Return `true` from this closure to raise `SIGTRAP`, or false to continue.
+     - returns: The source sequence that raises a debugger signal when one of the provided closures returns `true`.
+     */
+    public func breakpoint(
+        onNext: @escaping (Element) throws -> Bool = { _ in false },
+        afterNext: @escaping (Element) throws -> Bool = { _ in false },
+        onError: @escaping (Swift.Error) throws -> Bool = { _ in false },
+        afterError: @escaping (Swift.Error) throws -> Bool = { _ in false },
+        onCompleted: @escaping @autoclosure () throws -> Bool = false,
+        afterCompleted: @escaping @autoclosure () throws -> Bool = false,
+        onSubscribe: @escaping @autoclosure () -> Bool = false,
+        onSubscribed: @escaping @autoclosure () -> Bool = false,
+        onDispose: @escaping @autoclosure () -> Bool = false
+    ) -> Observable<Element> {
+        return Breakpoint(
+            source: self.asObservable(),
+            eventPredicate: {
+                switch $0 {
+                case let .next(element):
+                    return try onNext(element)
+                case let .error(error):
+                    return try onError(error)
+                case .completed:
+                    return try onCompleted()
+                }
+            },
+            afterEventPredicate:  {
+                switch $0 {
+                case let .next(element):
+                    return try afterNext(element)
+                case let .error(error):
+                    return try afterError(error)
+                case .completed:
+                    return try afterCompleted()
+                }
+            },
+            onSubscribe: onSubscribe(),
+            onSubscribed: onSubscribed(),
+            onDispose: onDispose()
+        )
+    }
+
+    /**
+      Raises a debugger signal upon receiving an error.
+      When the upstream Observable fails with an error, this operator raises the `SIGTRAP` signal, which stops the process in the debugger.
+      Otherwise, this Observable passes through values and completions as-is.
+      - seealso: [breakpointOnError operator on Combine](https://developer.apple.com/documentation/combine/publisher/3204689-breakpointonerror)
+      - returns: An Observable that raises a debugger signal upon receiving a failure.
+      */
+    public func breakpointOnError() -> Observable<Element> {
+        return Breakpoint(
+            source: self.asObservable(),
+            eventPredicate: {
```

Comment:
You can also do `$0.error != nil` if you want to shorten this. Leaving as-is is also 100% fine :)

Reply:
or maybe `@unknown default`? :)

Reply:
Sure, done